### PR TITLE
otk: add mtls into resolve ostree command

### PR DESCRIPTION
**Pull request by lzap**
SSIA

this allows resolution of pulp hosted repos

### CODE REVIEW

File: cmd/otk/osbuild-resolve-ostree-commit/main.go [hunk 1]
```diff
@@ -27,6 +27,16 @@ type Input struct {
 
 	// Whether to use RHSM secrets when resolving and fetching the commit.
 	RHSM bool `json:"rhsm,omitempty"`
+
+	// MTLS information. Will be ignored if RHSM is set.
```
Comment (mvo5) [inline cmd/otk/osbuild-resolve-ostree-commit/main.go:31]: There is a `main_test.go:TestMockResolve()` test, would it make sense to add the new inputs there (or create a new test analog to the existing one) to ensure we do not regress here?

**Comment by lzap** [comment]
Sure amended a test.

File: cmd/otk/osbuild-resolve-ostree-commit/main.go (hunk 1 above)
Comment (lzap) [reply cmd/otk/osbuild-resolve-ostree-commit/main.go:31]: Sure added but I do not understand why tests are failing now.

Comment (lzap) [reply cmd/otk/osbuild-resolve-ostree-commit/main.go:31]: Was looking at the wrong branch, rebased :-D

Comment (mvo5) [reply cmd/otk/osbuild-resolve-ostree-commit/main.go:31]: Well, tests are failing (locally too) :/
```console
$ go test -run TestResolver
--- FAIL: TestResolver (0.00s)
    main_test.go:90: 
        	Error Trace:	/home/mvogt/devel/osbuild/images/cmd/otk/osbuild-resolve-ostree-commit/main_test.go:90
        	Error:      	Not equal: 
        	            	expected: map[string]map[string]map[string]string{"tree":map[string]map[string]string{"const":map[string]string{"checksum":"9b1ea9a8e10dc27d4ea40545bec028ad8e360dd26d18de64b0f6217833a8443d", "ref":"test/ref/alpha", "url":"http://127.0.0.1:45097"}}}
        	            	actual  : map[string]map[string]map[string]string{"tree":map[string]map[string]string{"const":map[string]string{"checksum":"9b1ea9a8e10dc27d4ea40545bec028ad8e360dd26d18de64b0f6217833a8443d", "ref":"test/ref/alpha", "secrets":"org.osbuild.mtls", "url":"http://127.0.0.1:45097"}}}
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -2,5 +2,6 @@
        	            	  (string) (len=4) "tree": (map[string]map[string]string) (len=1) {
        	            	-  (string) (len=5) "const": (map[string]string) (len=3) {
        	            	+  (string) (len=5) "const": (map[string]string) (len=4) {
        	            	    (string) (len=8) "checksum": (string) (len=64) "9b1ea9a8e10dc27d4ea40545bec028ad8e360dd26d18de64b0f6217833a8443d",
        	            	    (string) (len=3) "ref": (string) (len=14) "test/ref/alpha",
        	            	+   (string) (len=7) "secrets": (string) (len=16) "org.osbuild.mtls",
        	            	    (string) (len=3) "url": (string) (len=22) "http://127.0.0.1:45097"
        	Test:       	TestResolver
    main_test.go:90: 
        	Error Trace:	/home/mvogt/devel/osbuild/images/cmd/otk/osbuild-resolve-ostree-commit/main_test.go:90
        	Error:      	Not equal: 
        	            	expected: map[string]map[string]map[string]string{"tree":map[string]map[string]string{"const":map[string]string{"checksum":"7433e1b49fb136d61dcca49ebe34e713fdbb8e29bf328fe90819628f71b86105", "ref":"test/ref/one", "url":"http://127.0.0.1:45097"}}}
        	            	actual  : map[string]map[string]map[string]string{"tree":map[string]map[string]string{"const":map[string]string{"checksum":"7433e1b49fb136d61dcca49ebe34e713fdbb8e29bf328fe90819628f71b86105", "ref":"test/ref/one", "secrets":"org.osbuild.mtls", "url":"http://127.0.0.1:45097"}}}
...
```

File: cmd/otk/osbuild-resolve-ostree-commit/main_test.go [hunk 2]
```diff
@@ -36,14 +39,25 @@ func createTestServer(refIDs map[string]string) *httptest.Server {
...
+	}
+	return server
 }
 
 func TestResolver(t *testing.T) {
 	require := require.New(t)
 	assert := assert.New(t)
 
-	repoServer := createTestServer(commitMap)
+	repoServer := createTestServer(commitMap, false)
```
Comment (mvo5) [inline cmd/otk/osbuild-resolve-ostree-commit/main_test.go:60]: (nitpick) I'm not a fan of bools in APIs like this, it makes it non-obvious what this "false" means. Which means one has to go back into the function do see what it means which will break the reading flow. the "low-tech" fix would be to just define a const above like "useMtls := false" and pass that. But looking at the code *maybe* a small refactor could also be an option, makes it slightly more indirect though:
```go
diff --git a/cmd/otk/osbuild-resolve-ostree-commit/main_test.go b/cmd/otk/osbuild-resolve-ostree-commit/main_test.go
index 11395309f..f3ff7695b 100644
--- a/cmd/otk/osbuild-resolve-ostree-commit/main_test.go
+++ b/cmd/otk/osbuild-resolve-ostree-commit/main_test.go
@@ -27,7 +27,7 @@ const TestCertDir = "../../../pkg/ostree/test_mtls_server"
 
 // Create a test server that responds with the commit ID that corresponds to
 // the ref.
-func createTestServer(refIDs map[string]string, mtls bool) *httptest.Server {
+func createTestHandler(refIDs map[string]string) *http.ServeMux {
        handler := http.NewServeMux()
        handler.HandleFunc("/refs/heads/", func(w http.ResponseWriter, r *http.Request) {
                reqRef := strings.TrimPrefix(r.URL.Path, "/refs/heads/")
@@ -38,26 +38,14 @@ func createTestServer(refIDs map[string]string, mtls bool) *httptest.Server {
                }
                fmt.Fprint(w, id)
        })
-
-       var server *httptest.Server
-       if mtls {
-               mtlss, err := test_mtls_server.NewMTLSServerInPath(handler, TestCertDir)
-               if err != nil {
-                       panic(err)
-               }
-
-               server = mtlss.Server
-       } else {
-               server = httptest.NewServer(handler)
-       }
-       return server
+       return handler
 }
 
 func TestResolver(t *testing.T) {
        require := require.New(t)
        assert := assert.New(t)
 
-       repoServer := createTestServer(commitMap, false)
+       repoServer := httptest.NewServer(createTestHandler(commitMap))
        defer repoServer.Close()
 
        url := repoServer.URL
@@ -95,7 +83,9 @@ func TestResolverMTLS(t *testing.T) {
        require := require.New(t)
        assert := assert.New(t)
 
-       repoServer := createTestServer(commitMap, true)
+       mtlsServer, err := test_mtls_server.NewMTLSServerInPath(createTestHandler(commitMap), TestCertDir)
+       require.NoError(err)
+       repoServer := mtlsServer.Server
        defer repoServer.Close()
 
        url := repoServer.URL
@@ -206,7 +196,7 @@ func TestResolverIDwithURL(t *testing.T) {
 
 func TestResolverErrors(t *testing.T) {
 
-       repoServer := createTestServer(commitMap, false)
+       repoServer := httptest.NewServer(createTestHandler(commitMap))
        defer repoServer.Close()
 
        type testCase struct {

```

**Review by mvo5 — COMMENTED** [review]
Thank you for working on this! Small ideas/suggestion inline, otherwise this looks very nice

**Comment by lzap** [comment]
Removed the boolean, rebased, fixed tests.

**Review by mvo5 — APPROVED** [review]
Thank you!

**Comment by lzap** [comment]
No other reviewers, adding to the merge queue. Thank you for your patience with me :)

**Pull request merged**